api/v1beta1/openstackmachine_types.go

```diff
@@ -119,6 +118,10 @@ type OpenStackMachineStatus struct {
 	// +optional
 	Ready bool `json:"ready"`
 
+	// InstanceID is the OpenStack instance ID for this machine.
+	// +optional
+	InstanceID optional.String `json:"instanceID,omitempty"`
```

Comment:
Have we considered adding it under the `Resolved` field?

Reply:
My first PR put it in `Resources`. This made sense, but I've decided against that for now. The reason is I think InstanceID is `fundamental` to the OpenStackMachine object and something an end user might be interested in (e.g. in `printcolumns`), but the other properties are not.

I have an idea that I might move actual machine creation into a separate controller with its own CRD, in which case `Resources` would move to the new controller. In that case I think we'd still want InstanceID to be on *this* object, so I decided to put it here instead.

None of which precludes adding it to Resources in the future too if it turns out that still makes sense.